tests/difference/core/quint_model/ccv.qnt

```diff
@@ -0,0 +1,619 @@
+    // at blocks over its entire existence.
+    // Voting powers should be ordered by recency in descending order.
+    var votingPowerHistories: Chain -> List[ValidatorSet]
+
+    // the current validator set on each chain.
+    // this will be included in the next block, but might not be final yet,
+    // e.g. there may be more modifications in the current block.
+    var runningValidatorSet: Chain -> ValidatorSet
+
+    // the current timestamp for each chain
+    var curChainTimes: Chain -> Timestamp
+
+    // stores, for each chain, its current status -
+    // unused, running, or stopped
+    var consumerStatus: Chain -> str
+
+    // --CHANNELS
+    // Stores, for each consumer chain, the list of packets that have been sent to the provider chain
+    // and have not been received yet.
+    var outstandingPacketsToProvider: Chain -> List[VSCMaturedPacket]
+
+    // Stores, for each consumer chain, the list of packets that have been sent to the consumer chain
+    // and have not been received yet.
+    var outstandingPacketsToConsumer: Chain -> List[VSCPacket]
+
+
+    // --CONSUMER STATE
+    // Stores the maturation times for VSCPackets received by consumers
+    var maturationTimes: Chain -> (VSCPacket -> Timestamp)
+
+    // --PROVIDER STATE
+    // the set of VSCMaturedPackets received by the provider chain
+    var receivedMaturations: Set[VSCMaturedPacket]
+
+    // stores which VSC Packets have been sent to compare with receivedMaturations to detect timeouts due to non-responsiveness
+    var sentVSCPackets: Chain -> Set[VSCPacket]
+
+    // stores whether, in this step, the validator set considered to be changed.
+    // this is needed because the validator set might be considered to have changed, even though
+    // it is still technically identical at our level of abstraction, e.g. a validator power change on the provider
+    // might leave the validator set the same because a delegation and undelegation cancel each other out.
+    var providerValidatorSetChangedInThisBlock: bool
+
+
+    // UTILITY FUNCTIONS & ACTIONS
+    def wasValidatorSetOnProvider(validatorSet: ValidatorSet): bool = {
+        votingPowerHistories.get(ProviderChain).toSet().exists(
+            historicalValSet => historicalValSet == validatorSet
+        )
+    }
+
+    def getCurrentValidatorSet(chain: Chain): ValidatorSet = 
+        votingPowerHistories.get(chain).head()
+
+    def getUpdatedValidatorSet(oldValidatorSet: ValidatorSet, validator: Node, newVotingPower: int): ValidatorSet =
+        if (newVotingPower > 0) 
+        oldValidatorSet.put(validator, newVotingPower)
+        else
+        oldValidatorSet.mapRemove(validator)
+
+    // returns true if the consumer has timed out and should be dropped
+    def consumerTimedOut(consumer: Chain): bool =
+        any {
+            // either a package from provider to consumer has timed out
+            outstandingPacketsToConsumer.get(consumer).select(
```

Comment:
Wouldn't this be more efficient to just look at the oldest packet in the list? Or we do not remove packets once they arrive?